platform/app/src/routes/Mode/Mode.tsx

```diff
@@ -124,6 +124,15 @@ export default function ModeRoute({
   const params = useParams();
   // The URL's query search parameters where the keys casing is maintained
   const query = useSearchParams();
+
+  mode?.onModeInit({
```

Comment:
@brunoalvesdefaria was thinking of adding another hook, can you sync with him to see what was his use case maybe we merge both of your hooks into one?